google/resource_dataproc_job.go

```diff
@@ -0,0 +1,1068 @@
+package google
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"google.golang.org/api/dataproc/v1"
+)
+
+func resourceDataprocJob() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceDataprocJobCreate,
+		Read:   resourceDataprocJobRead,
+		Delete: resourceDataprocJobDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+
+			"project": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			// Ref: https://cloud.google.com/dataproc/docs/reference/rest/v1/projects.regions.jobs#JobReference
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "global",
+				ForceNew: true,
+			},
+
+			// If a job is still running, trying to delete a job will fail. Setting
+			// this flag to true however will force the deletion by first cancelling
+			// the job and then deleting it
+			"force_delete": {
+				Type:     schema.TypeBool,
+				Default:  false,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"reference": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"job_id": {
```

Comment:
I'm not going to block on this, but I'd love to see a ValidateFunc here (I think the regex you want is `^[a-zA-Z0-9_-]{1,100}$`)